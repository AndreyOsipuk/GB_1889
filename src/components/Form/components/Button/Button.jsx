import React from 'react';

export const Button = ({ disabled, onButtonClick }) => {
  return (
    <button type="submit" disabled={disabled} onClick={onButtonClick}>
      click
    </button>
  );
};
