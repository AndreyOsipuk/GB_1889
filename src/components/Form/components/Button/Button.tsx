import React, { FC } from 'react';
// import { Button as ButtonUI } from '@mui/material';
// import './Button.scss';

interface ButtonProps {
  disabled: boolean;
  onButtonClick?: () => void;
}

export const Button: FC<ButtonProps> = ({ disabled, onButtonClick }) => (
  <button
    // variant="contained"
    type="submit"
    disabled={disabled}
    onClick={onButtonClick}
  >
    click
  </button>
);
