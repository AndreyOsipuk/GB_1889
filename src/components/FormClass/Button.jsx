import React, { Component } from 'react';

export class Button extends Component {
  render() {
    return <button onClick={this.props.click}>{this.props.name}</button>
  }
}
