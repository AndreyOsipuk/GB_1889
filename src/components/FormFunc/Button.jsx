import React from 'react';

export const Button = (props) => {
  return <button onClick={props.click}>{props.name}</button>;
};
