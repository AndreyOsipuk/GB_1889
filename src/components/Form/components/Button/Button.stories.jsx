import React from 'react';

import { Button } from './Button';

export default {
  title: 'MyComponents/Button',
  component: Button,
  argTypes: {
    onButtonClick: { action: 'click' },
  },
};

const Template = (args) => <Button {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  disabled: false,
};

export const Disabled = Template.bind({});
Disabled.args = {
  disabled: true,
};
