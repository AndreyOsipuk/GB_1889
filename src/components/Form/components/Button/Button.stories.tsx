import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';

import { Button } from './Button';

export default {
  argTypes: {
    onButtonClick: { action: 'click' },
  },
  component: Button,
  title: 'MyComponents/Button',
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  disabled: false,
};

export const Disabled = Template.bind({});
Disabled.args = {
  disabled: true,
};
