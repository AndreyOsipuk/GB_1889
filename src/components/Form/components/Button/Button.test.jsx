import React from 'react';
import { Button } from './Button';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import { waitFor } from '@storybook/testing-library';
describe('Button', () => {
  it('render component', () => {
    render(<Button />);
  });

  it('render with snapshot', () => {
    const { asFragment } = render(<Button />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('render component with text', () => {
    render(<Button />);
    expect(screen.getByText(/click/)).toBeInTheDocument();
  });

  it('render multiply components', () => {
    render(
      <>
        <Button />
        <Button />
      </>
    );

    expect(screen.queryAllByRole('button').length).toBe(2);
  });

  it('button is disabled', () => {
    render(<Button disabled />);

    expect(screen.getByText('click')).toBeDisabled();
  });

  // it('button have style background red', () => {
  //   render(<Button />);

  //   expect(screen.getByText('click')).toHaveStyle({ backgroundColor: 'red' });
  // });

  it('button click with userEvent', async () => {
    const mockHandler = jest.fn();

    render(<Button onButtonClick={mockHandler} />);
    await userEvent.click(screen.getByText(/click/));
    expect(mockHandler).toBeCalledTimes(1);
  });

  it('button async click', async () => {
    const mockHandler = jest.fn();
    render(<Button onButtonClick={() => setTimeout(mockHandler, 1000)} />);

    await userEvent.click(screen.getByText(/click/));

    await waitFor(() => expect(mockHandler).toHaveBeenCalledTimes(1), {
      timeout: 1100,
    });
  });

  it('test exapmle', async () => {
    const onChange = jest.fn();
    render(<input type="checkbox" onChange={onChange} />);
    const checkbox = screen.getByRole('checkbox');
    await userEvent.dblClick(checkbox);
    expect(onChange).toHaveBeenCalledTimes(2);
    expect(checkbox).not.toBeChecked();
  });
});
