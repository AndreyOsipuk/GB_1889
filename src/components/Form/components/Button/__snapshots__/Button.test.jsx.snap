// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render with snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="submit"
  >
    click
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
